### DISCUSSION TITLE
Bazarr with Plex Debrid for subtitles?

### DISCUSSION THREAD

**Discussion by arjanterheegde** [itsToggle/plex_debrid#364] [Q&A]
Is it possible to run Bazarr without Sonarr and Radarr? I want to use it so my Plex Media “downloaded” with Plex Debrid will automatically provided with Dutch subtitles.

Or do you guys recommend any other tool? Thanks in advance!

**Comment by arjanterheegde** [comment]
I've tried Plex sub downloader but it won't add any subtitles. Will try tonight again.
https://github.com/replicatingTrouts/plex-sub-downloader

**Reply by sasagr** [reply]
> I've tried Plex sub downloader but it won't add any subtitles. Will try tonight again. https://github.com/replicatingTrouts/plex-sub-downloader

this will not work as it is, simply because with plex_debrid we dont have the permission to write files in the download folder. To make this script works it would need some changes. I can list them to you here if you want work on them.
- create local folders of movie and tvshows, same names that you see in plex_debrid mount
- change the sub downloader script to create a local folder for each movie/tv shows every time a new media is added and subtitle is found, the local folder has to be the same name of what you will see in plex_debrid mount
- create an union mount between you local folder and plex_debrid mount

Using this parallel local structure you will be able to save locally what ever you want and then because you plex library will point to the union mount, you will have all you need.

I don't see other better solutions at the moment

**Reply by arjanterheegde** [reply]
Thanks! I will definitely try this and let you guys know.

So the option subtitle_destination => metadata doesn't work?

![image](https://user-images.githubusercontent.com/6838586/214816146-193fc347-d4e9-489e-ae3a-d855096448b1.png)

**Reply by sasagr** [reply]
I m very sorry I mislead you... I just confused this script with a couple of others I also use (hide_spoiler and plex_utils) where I needed the above mentioned workaround to make them work. For the sub downloader I m actually using what you are suggesting. 
"subtitle_destination": "metadata" in the json file.
It works well for me but I remember I did a couple of modifications there as well in order to run not only when the media is added but also when it is played (in case the subtitle is not available when added) and also to search for subtitles even if embedded subtitle exist but not if an opensubtitles or external exists already). If I remember correct the original script does not look for subtitles if it already exist (embedded or not).
I hope this is not confusing :)
I did some work on them a few months ago so I dont remember all very well... should go through them again. One thing I rememeber that I had issues with other providers rather than opensubtitles vip.

**Reply by arjanterheegde** [reply]
Can you share the modifications if you still have them :)?

**Reply by sasagr** [reply]
sure. I will give you the mod script that you can install with pip
run this in the directory of the file
pip3 install plex_sub_downloader-0.1.14.mod.tar.gz

[plex_sub_downloader-0.1.14.mod.tar.gz](https://github.com/itsToggle/plex_debrid/files/10509328/plex_sub_downloader-0.1.14.mod.tar.gz)

**Reply by arjanterheegde** [reply]
Ur amazing! I will try tonight and let you know if this works.

**Reply by sasagr** [reply]
no worries

dont forget to edit the config.json file and add the webhook in your plex

{
    "plex_base_url": "http://127.0.0.1:32400",
    "plex_auth_token": "xxxx",
    "webhook_host": "0.0.0.0",
    "webhook_port": 5000,
    "subtitle_destination": "metadata",
    "languages": [
        "eng"
    ],
    "subtitle_providers": [
        "opensubtitlesvip"
    ],
    "subtitle_provider_configs": {
        "opensubtitlesvip": {
            "username": "xxx",
            "password": "xxxx"
        }
    },
    "log_level": 20
}

**Reply by arjanterheegde** [reply]
Do you know where i can find the Python logs for plex-sub-download? It's my first time using Python, but i wan't to learn it.
<img width="929" alt="image" src="https://user-images.githubusercontent.com/6838586/214904247-ec4d17a7-824b-488f-9e37-cccea63cd9b1.png">

**Reply by sasagr** [reply]
I don’t think you will see much from the main logs of your machine. The best debug you can get from Plex downloader is if you use the debug flag and you change the debug value in the config file to 10 if I m not wrong. By the way are you adding the config path to your command? Show me the full command you are executing

**Reply by arjanterheegde** [reply]
plex_sub_downloader --config config.json start-webhook --debug

When i play or stop something in plex the only thing i see is:
127.0.0.1 - - [26/Jan/2023 18:49:46] "POST / HTTP/1.1" 404 -
127.0.0.1 - - [26/Jan/2023 18:49:50] "POST / HTTP/1.1" 404 -

**Reply by sasagr** [reply]
I m not sure it is finding your config file. It should show you the main setup of your config.  Did u run the configtest line?

**Reply by arjanterheegde** [reply]
Yes, and it can find the config. Don't know what i'm doing wrong haha.
<img width="921" alt="image" src="https://user-images.githubusercontent.com/6838586/214913900-92bfda4f-255b-47e7-b127-2436b74bad01.png">

**Reply by sasagr** [reply]
I guess you added the webhook address in Plex

**Reply by arjanterheegde** [reply]
Yes
<img width="1058" alt="image" src="https://user-images.githubusercontent.com/6838586/214922004-bae14f59-5b17-4ae9-a434-c73c2c6f4916.png">

**Reply by sasagr** [reply]
Add webhook at the end
http://192.168.1.100:5000/webhook

**Reply by arjanterheegde** [reply]
Thanks for your time, it works now!

**Reply by roisec** [reply]
Are you able to do that without webhook?and without Plex press. 
I meant maybe integrating the scraping phase , with debrid_plex code , download a subtitles and than start download the movie.. 
Is it possible?

**Reply by sasagr** [reply]
this solution needs webhook and this is available only with plex pass AFAIK

**Reply by roisec** [reply]
@sasagr Yes but it can be done through the plex_debrid service and download the subtitles after the scraping process.
I think it may useful for everyone because it's free..😄

**Comment by roisec** [comment]
@sasagr can you explain please what do you mean union mount? How can I do that?

**Reply by sasagr** [reply]
you will find better explanation here, https://rclone.org/union/

**Reply by roisec** [reply]
@sasagr thank you. 
Can you share how do you configure the union volume, and how did it works with Plex? (Because it's different directory)
Thanks!!

**Reply by sasagr** [reply]
I use rclone union for the purpose of having a local destination for those scripts that produce files that need to be store locally in the same directory of the media. Because rclone debrid is READ only then you need to create a parallel local structure that can be written. Rclone union will "unify" the local writable structure with reclone debrid mount and you can feed it to plex. The logic is easy, to do it it a bit more tricky and I have done it customized with my needs. It would take me very long time to produce something that could work for you and unfortunately I dont have it :( 
In any case as I already stated in a previous post you dont need it for the subtitles script mentioned here. You can set it up to store them in the metadata.

**Reply by roisec** [reply]
@sasagr 
I created union type storage and added the realdebrid: and local folder...
when I run cmount the union storage I got:
![image](https://user-images.githubusercontent.com/68992144/218107795-a1dfae43-7cbf-45e2-9bd8-f69762160ac8.png)
but got this problem...
do you know what is the problem?

**Comment by arjanterheegde** [comment]


**Reply by arjanterheegde** [reply]
Anyone?

**Reply by arjanterheegde** [reply]
Still hoping... Anyone? 😄
